go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -383,9 +381,11 @@ func (n *NodeController) EnsureHybridOverlayBridge(node *kapi.Node) error {
 	}
 	n.drMAC = portMAC
 
-	// n.drIP is always 3rd address in the subnet
-	hybridOverlayIfAddr := util.GetNodeHybridOverlayIfAddr(subnet)
-	n.drIP = hybridOverlayIfAddr.IP
+	hybridOverlayIFAddress, ok := node.Annotations[hotypes.HybridOverlayPortAddress]
+	if !ok {
+		return fmt.Errorf("hybrid overlay not initialized on %s, it was not assigned an interface address", node.Name)
+	}
+	n.drIP = net.ParseIP(hybridOverlayIFAddress)
```

Comment:
NIT: may want to return an error if net.ParseIP returns nil

Reply:
done